src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/NumericClientModelValidatorProvider.cs

```diff
@@ -27,7 +27,11 @@ public void GetValidators(ClientValidatorProviderContext context)
                 typeToValidate == typeof(double) ||
                 typeToValidate == typeof(decimal))
             {
-                context.Validators.Add(new NumericClientModelValidator());
+                context.Results.Add(new ClientValidatorItem
+                {
+                    Validator = new NumericClientModelValidator(),
+                    IsReusable = true
+                });
```

Comment:
This need to still work when the validators are partially cached. This will just keep adding more and more instances of this validator as-is.